pkg/agent/prommetrics.go

```diff
@@ -344,9 +347,22 @@ func WrapHistogramVec(hist *prometheus.HistogramVec) revsource.ObserveCallback {
 }
 
 type PerVMMetrics struct {
+	// activeMu and activeVMs exist to track the set of VMs currently represented in the metrics, so
+	// that when we set the desired CU from internal information, we can check whether the VM still
+	// exists.
+	// Otherwise it's not possible to prevent data races that would result in leaking metric labels.
+	activeMu  sync.Mutex
```

Comment:
Why did this become needed only now, how `desiredCU` is different from other metrics we already had?

Reply:
IIRC, it wasn't needed before because the metrics were all derived from the state of the VM, whereas `desiredCU` is derived from our internal state.